src/main/java/com/splunk/hecclient/Indexer.java

```diff
@@ -192,14 +188,13 @@ public CloseableHttpResponse run() {
         return readAndCloseResponse(resp);
     }
 
-    private Subject getSubject() throws LoginException {
+    private void getSubject() throws LoginException {
         Set<Principal> princ = new HashSet<Principal>(1);
         princ.add(new KerberosPrincipal(hecConfig.kerberosUser()));
         Subject sub = new Subject(false, princ, new HashSet<Object>(), new HashSet<Object>());
         LoginContext lc = new LoginContext("", sub, null, config);
         lc.login();
-        Subject serviceSubject = lc.getSubject();
-        return serviceSubject;
+        serviceSubject = lc.getSubject();
```

Comment:
why we are not returning the `Subject` object?
by testing convention, the getter method should return the object for which the getter has been created.